### PR TITLE
Add join table for visits

### PR DESCRIPTION
**Pull request by MostafaShahat2021**
## In this PR:
-⭐️Use primary key & foreign key mechanisms for joining tables.
-⭐️Query multiple tables.
-⭐️Make complex queries that answer analytical questions.
-⭐️Use different types of relationships between tables.
-⭐️vets table :
image

image

-⭐️specializations join table : 
image

image

-⭐️visits join table :
image

image

### -⭐️Queries : 
- Who was the last animal seen by William Tatcher? 
image

- ⭐️How many different animals did Stephanie Mendez see?
image

-⭐️List all vets and their specialties, including vets with no specialties
image

-⭐️List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
image

- ⭐️What animal has the most visits to vets?
image

-⭐️Who was Maisy Smith's first visit?
image

-⭐️Details for most recent visit: animal information, vet information, and date of visit.
image

-⭐️How many visits were with a vet that did not specialize in that animal's species?
image

-⭐️What specialty should Maisy Smith consider getting? Look for the species she gets the most.
image

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -138,4 +138,81 @@ FROM owners o
...
+ORDER BY v.visit_date DESC
+LIMIT 1;
+
+-- How many visits were with a vet that did not specialize in that animal's species?
+SELECT COUNT(*) AS num_visits_without_specialization
+FROM visits v
+JOIN animals a ON v.animal_id = a.id
+JOIN vets vt ON v.vet_id = vt.id
+LEFT JOIN specializations s ON vt.id = s.vet_id AND a.species_id = s.species_id
+WHERE s.vet_id IS NULL;
```
Comment (mckent05) [inline queries.sql:203]: - You have done well so far, but from your screenshot to the query, the answer for this query is incorrect, I suggest that you rerun this query and get the right answer. Please fix this and attach the correct screenshot to your  🙏. The answer you should be getting is 9.

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @MostafaShahat2021, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍
- Good job adding join table for visits 👍 👍


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (MostafaShahat2021) [reply queries.sql:203]: Hi @mckent05 👋, 
Hope all is well with you, I see that you Approved my project, but in the dashboard, the status shows as CHANGES REQUIRED, Could you please fix the status in the dashboard so I can submit the project
Thanks a lot for your efforts in reviewing my code

Comment (mckent05) [reply queries.sql:203]: Hello @MostafaShahat2021, I have checked thoroughly and realised that the answer to the query in your screenshot is the right one. I am sorry about the inconveniences, please submit your project and it should be approved 🙂🙏

Comment (MostafaShahat2021) [reply queries.sql:203]: Hi @mckent05 Thanks a lot 💯 for your efforts in reviewing my code 👍

**Review by t-yanick — APPROVED** [review]
# STATUS: MILESTONE APPROVED :1st_place_medal: 

Hi @MostafaShahat2021 , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Excellent communication with reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by MostafaShahat2021** [comment]
HI @t-yanick 👋, 
Thanks a lot 💯 for your efforts in reviewing my code, I really appreciate it 😄

**Pull request merged**